### PR TITLE
Streamlit Frontend for displaying the NCBI Document Summaries

### LINKED ISSUE #13: Add Front-End for displaying the NCBI document summaries

### PR DESCRIPTION
**Pull request by aniketfadia96**
I have made the following changes in this PR - 

1. Combine input fields for `database` and `search term` inside a Streamlit Form
2. Display the results after submitting the Form from Step 1 in an `aggrid table`
3. Provide additional filtering UI to further filter the results of Step 2

Note
* I have tried to modularize the Streamlit application into separate packages serving below high-level functionality -
1. `common`: Basic Functions/Constants shared across the Streamlit app. I am already using a `utils` folder for the entire `neglected-diagnostics` package, so I went ahead with `common` as the name of the package.
2. `frontend`: Custom Frontend components which are not directly supported by Streamlit. For example, `Aggrid Table` and `Filter Dataframe UI`.

Closes #13

### CODE REVIEW

File: src/app/common/preprocessing.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import streamlit as st
+
+from app.common.constants import NCBI_DF
+
+
+def format_ncbi_summary():
```
Comment (lsetiawan) [inline src/app/common/preprocessing.py:6]: Output typehint here

File: src/app/common/setup.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import pandas as pd
+import streamlit as st
+
+from app.common.constants import NCBI_DF, NCBI_DF_FILTER, NCBI_SUMMARY_FORM
+
+
+def initialize():
```
Comment (lsetiawan) [inline src/app/common/setup.py:7]: Return typehint

**Review by lsetiawan — APPROVED** [review]
Minor comments, looking good 😄

File: src/app/common/preprocessing.py (hunk 1 above)
Comment (aniketfadia96) [reply src/app/common/preprocessing.py]: Added Type Hint

File: src/app/common/setup.py (hunk 2 above)
Comment (aniketfadia96) [reply src/app/common/setup.py]: Added Type Hint

**Pull request merged**